src/pages/components/toggle.tsx

```diff
@@ -14,15 +15,29 @@ import ComponentExample from 'components/ComponentExample';
 import { Heading, Text, Paragraph } from 'components/ResponsiveComponents';
 
 const Page: React.FC = (): React.ReactElement => {
-  const [isChecked, setChecked] = useState(false);
+  const [isButtonChecked, setButtonChecked] = useState(false);
+  const [isInputChecked, setInputChecked] = useState(false);
 
-  const getAriaLabelText = (t: Function): string =>
-    t(`toggle.state.${isChecked ? 'on' : 'off'}`, {
-      name: t('toggle.label'),
+  const getAriaLabelText = (t: Function, inputVariant = false): string => {
+    const componentVariantState = inputVariant
+      ? isInputChecked
+      : isButtonChecked;
+
+    const componentVariantLabel = inputVariant
+      ? 'toggle.label.input'
+      : 'toggle.label.button';
+
+    return t(`toggle.state.${componentVariantState ? 'on' : 'off'}`, {
+      name: t(componentVariantLabel),
     });
+  };
```

Comment:
Using same function for both variations.
What are you opinion for this kind of logic done in same function for both cases?

Reply:
I already dislike the whole concept of having two separate variants sharing the same codebase.

Reply:
Should we then separate the variants to own files to keep it more readable?

Reply:
Not sure, this issue originates from the ui-comps repo. How much duplication would this introduce?

Reply:
I guess either version works, one function for both variants or separate functions.